src/manager/monitor.c

```diff
@@ -374,89 +395,381 @@ static int monitor_method_unsubscribe(sd_bus_message *m, void *userdata, UNUSED
         return sd_bus_reply_method_return(m, "");
 }
 
+/*******************************************************
+ ********* org.eclipse.bluechi.Monitor.AddPeer *********
+ *******************************************************/
+
+static MonitorPeer *monitor_add_peer(Monitor *monitor, const char *peer_name) {
+        static uint32_t next_peer_id = 0;
+
+        MonitorPeer *peer = malloc0(sizeof(MonitorPeer));
+        if (peer == NULL) {
+                return NULL;
+        }
+        peer->name = strdup(peer_name);
+        if (peer->name == NULL) {
+                free(peer);
+                return NULL;
+        }
+        peer->id = ++next_peer_id;
+
+        LIST_APPEND(peers, monitor->peers, peer);
+
+        bc_log_debugf("Added monitor peer '%s' to monitor '%s'", peer_name, monitor->object_path);
+        return peer;
+}
+
+static int monitor_method_add_peer(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        Monitor *monitor = userdata;
+
+        const char *peer_name = NULL;
+
+        int r = sd_bus_message_read(m, "s", &peer_name);
+        if (r < 0) {
+                return sd_bus_reply_method_errorf(
+                                m,
+                                SD_BUS_ERROR_INVALID_ARGS,
+                                "Invalid argument for the peer name: %s",
+                                strerror(-r));
+        }
+
+        if (!sd_bus_service_name_is_valid(peer_name)) {
```

Comment:
Really we should validate that it is a unique bus id. I.e. not accept a service name here.

Reply:
I mean, we should only accept ":1.14504", i.e. a physical connection, not a service name. Both can be used as targets of a message, but the destination peer will change over time for a service name as the owner changes.

Reply:
Think of these like unique-id == ip address, service-name == dns name.